ML- Filter lists - error message seen on clicking refresh button and also on creating a New filter list

**Kibana version:** 7.2.0 BC3

**Elasticsearch version:**7.2.0 BC3

**Server OS version:** **cloud- staging**

**Browser version:** Chrome latest 

**Browser OS version:**

**Original install method (e.g. download page, yum, from source, etc.):** 7.2.0 BC3

**Describe the bug:** When you navigate to Machine Learning > Filter Lists - and hit refresh - there is a pop up which occurs  with an error message ` An error occurred loading the filter lists`  This is the same error message seen while you try to create a new Filter list . Unable to create and save a new Filter list. 

<img width="1676" alt="Screen Shot 2019-06-03 at 5 35 15 AM" src="https://user-images.githubusercontent.com/11466284/58802729-bea2ca00-85c2-11e9-82af-7164a5769497.png">

<img width="1660" alt="Screen Shot 2019-06-03 at 5 49 43 AM" src="https://user-images.githubusercontent.com/11466284/58803042-6ae4b080-85c3-11e9-9456-a0682fc9fafd.png">



Console logs: 
```
ml.bundle.js:3 GET https://c8e2cb6a16554611b6d7328e625f7d18.us-east-1.aws.staging.foundit.no:9243/api/ml/filters/_stats 400 (Bad Request)
(anonymous) @ ml.bundle.js:3
http @ ml.bundle.js:3
filtersStats @ ml.bundle.js:3
(anonymous) @ ml.bundle.js:3
onClick @ ml.bundle.js:3
ca @ vendors.bundle.dll.js:233
ja @ vendors.bundle.dll.js:233
ka @ vendors.bundle.dll.js:233
wa @ vendors.bundle.dll.js:233
Aa @ vendors.bundle.dll.js:233
ya @ vendors.bundle.dll.js:233
Da @ vendors.bundle.dll.js:233
Ad @ vendors.bundle.dll.js:233
Gi @ vendors.bundle.dll.js:233
Kb @ vendors.bundle.dll.js:233
Dd @ vendors.bundle.dll.js:233
(anonymous) @ vendors.bundle.dll.js:233
exports.unstable_runWithPriority @ vendors.bundle.dll.js:242
Ii @ vendors.bundle.dll.js:233
Cd @ vendors.bundle.dll.js:233
ml.bundle.js:3 Error loading list of filters: {statusCode: 400, error: "Bad Request", message: "[query_shard_exception] No mapping found for [filt…uid="aFu__YilQsGgBnUh_bhH3w" & index=".ml-meta" }"}
```

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/AbstractTransportGetResourcesAction.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/AbstractTransportGetResourcesAction.java
@@ -30,6 +30,7 @@ import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.elasticsearch.search.SearchHit;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.SortBuilders;
 import org.elasticsearch.transport.TransportService;
 import org.elasticsearch.xpack.core.action.util.ExpandedIdsMatcher;
 import org.elasticsearch.xpack.core.action.util.QueryPage;
@@ -70,7 +71,10 @@ public abstract class AbstractTransportGetResourcesAction<Resource extends ToXCo
     protected void searchResources(AbstractGetResourcesRequest request, ActionListener<QueryPage<Resource>> listener) {
         String[] tokens = Strings.tokenizeToStringArray(request.getResourceId(), ",");
         SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
-            .sort(request.getResourceIdField())
+            .sort(SortBuilders.fieldSort(request.getResourceIdField())
+                // If there are no resources, there might be no mapping for the id field.
+                // This makes sure we don't get an error if that happens.
+                .unmappedType("long"))
             .query(buildQuery(tokens, request.getResourceIdField()));
         if (request.getPageParams() != null) {
             sourceBuilder.from(request.getPageParams().getFrom())

--- a/x-pack/plugin/ml/qa/ml-with-security/build.gradle
+++ b/x-pack/plugin/ml/qa/ml-with-security/build.gradle
@@ -47,6 +47,7 @@ integTestRunner {
     'ml/filter_crud/Test invalid param combinations',
     'ml/filter_crud/Test non-existing filter',
     'ml/filter_crud/Test update filter given remove item is not present',
+    'ml/filter_crud/Test get all filter given index exists but no mapping for filter_id',
     'ml/get_datafeed_stats/Test get datafeed stats given missing datafeed_id',
     'ml/get_datafeeds/Test get datafeed given missing datafeed_id',
     'ml/jobs_crud/Test cannot create job with existing categorizer state document',

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/ml/filter_crud.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/ml/filter_crud.yml
@@ -328,3 +328,18 @@ setup:
       ml.get_filters: {}
   - match: { count: 0 }
   - match: { filters: [] }
+
+---
+"Test get all filter given index exists but no mapping for filter_id":
+
+  - do:
+      indices.delete:
+        index: ".ml-meta"
+  - do:
+      indices.create:
+        index: ".ml-meta"
+
+  - do:
+      ml.get_filters: {}
+  - match: { count: 0 }
+  - match: { filters: [] }

